src/riak_cs_s3_auth.erl

```diff
@@ -77,6 +79,18 @@ authenticate(User, Signature, RD, _Ctx) ->
 %% Internal functions
 %% ===================================================================
 
+%% Idintify user by query string.
+%% Currently support signature v2 only, does NOT support signature v4.
+identify_by_query_string(RD) ->
+    {wrq:get_qs_value(?QS_KEYID, RD), wrq:get_qs_value(?QS_SIGNATURE, RD)}.
+
+parse_auth_header("AWS4-HMAC-SHA256 " ++ String) ->
+    case riak_cs_config:auth_v4_enabled() of
+        true ->
+            parse_auth_v4_header(String, undefined, []);
+        _ ->
+            {failed, {auth_not_supported, "AWS4-HMAC-SHA256"}}
```

Comment:
Ah, this message comes out to response to clients. not supported is correct message as a whole.